cronJobs/Multipay-Airdrop-CronJob/airdropData/releaseAirdrops.ts

```diff
@@ -0,0 +1,117 @@
+import fetch from "cross-fetch";
+import { releaseAirdrops } from "../utils/contracts/dropper";
+import { execute, makePromise } from "apollo-link";
+import { createHttpLink } from "apollo-link-http";
+import gql from "graphql-tag";
+import { Chain, CHAINS } from "../utils/chain";
+import { getCurrentTimeInSecond, getDropperSubgraphURL } from "../utils/util";
+import { AirdropInfo } from "./interface";
+
+async function callReleaseAirdrops(
+  chain: Chain,
+  froms: string[][],
+  indexes: string[][],
+  droppperContract: string[]
+) {
+  const status = await releaseAirdrops(chain, froms, indexes, droppperContract);
+  if (status == 0x1) {
+    return true;
+  }
+  return false;
+}
+
+export default async function (chainId: number) {
+  console.log(`Initializing Airdrop process on chain ${chainId}...\n`);
+  const chain = CHAINS.find((c) => c.id == chainId);
+  if (chain == undefined) {
+    console.error(`Chain with ID ${chainId} not found.`);
+    return;
+  }
+  try {
+    const currentTimestamp = await getCurrentTimeInSecond();
+    const perPage = 100;
+    let dataCount = 0;
+
+    while (1) {
+      const filteredAirdropData = await filterAirdrops(
+        chain,
+        perPage,
+        dataCount,
+        currentTimestamp,
+        "lte"
+      );
+
+      const airdropInfos: AirdropInfo[] =
+        filteredAirdropData?.data?.hectorDropperContracts;
+
+      let contracts = airdropInfos
+        .filter((contract) => contract.airdrops.length > 0)
+        .map((contract) => contract.address);
+
+      let froms: string[][] = airdropInfos
+        .filter((contract) => contract.airdrops.length > 0)
+        .map((contract) =>
+          contract.airdrops.map((airdrop) => airdrop.from.address)
+        );
+      let indexes: string[][] = airdropInfos
+        .filter((contract) => contract.airdrops.length > 0)
+        .map((contract) => contract.airdrops.map((airdrop) => airdrop.index));
+
+      let usersLength = 0;
+      for (let i = 0; i < froms.length; i++) {
+        usersLength += froms[i].length;
+      }
+
+      if (contracts.length > 0 && usersLength > 0) {
+        console.log(`Initializing Airdrop Release on chain ${chainId}...\n`);
+
+        const isSuccess = await callReleaseAirdrops(
```

Comment:
@michaelnguyen102, the web3 call can be processed once all entity is read so after while(1) is done

Reply:
@bejaxer because of the paging, web3 call to release airdrop will be called per 100 records to make sure we have a manageable load. 
While reading all entities is possible, the risk of failure from releasing airdrops for all entities is likely to go up. Unless we are using our own RPC which we can control the traffic, it's hard to estimate the traffic when running on public rpc.